extract_token_from_browser.py

```diff
@@ -1,64 +1,176 @@
 
+from pathlib import Path
+from Crypto.Cipher import AES
+from Crypto.Protocol.KDF import PBKDF2
+
+
+class AESCipher:
+
+    def __init__(self, key):
+        self.key = key
+
+    def decrypt(self, text):
+        cipher = AES.new(self.key, AES.MODE_CBC, IV=(' ' * 16))
+        return self._unpad(cipher.decrypt(text))
+    
+    def _unpad(self, s):
+        return s[:-ord(s[len(s) - 1:])]
+
+
 parser = argparse.ArgumentParser(
     description="Extract Slack tokens from the browser files"
 )
 parser.add_argument(
     "browser", help="Which browser to extract from", metavar="<browser>"
 )
+parser.add_argument(
+    "--token", help="Extracted token", metavar="<token>", nargs='?'
+)
+parser.add_argument(
+    "--profile", help="Profile to look up cookies for", metavar="<profile>", nargs='?'
+)
 args = parser.parse_args()
 
-if args.browser != "firefox":
-    print("Currently only firefox is supported by this script", file=sys.stderr)
+if args.browser not in ["firefox", "firefox-snap", "chrome", "chrome-beta"]:
+    print("Currently only firefox. firefox-snap, chrome, "
+          "chrome-beta are supported by this script", file=sys.stderr)
     sys.exit(1)
 
 if sys.platform.startswith("linux"):
-    firefox_path = Path.home().joinpath(".mozilla/firefox")
+    iterations = 1
+    if args.browser == "firefox-snap":
+        browser_data = Path.home().joinpath(
+            "snap/firefox/common/.mozilla/firefox")
+    elif args.browser == "firefox":
+        browser_data = Path.home().joinpath(".mozilla/firefox")
+    else:
+        browser_data = Path.home().joinpath(".config/google-%s" % args.browser)
 elif sys.platform.startswith("darwin"):
-    firefox_path = Path.home().joinpath("Library/Application Support/Firefox/Profiles")
+    iterations = 1003
+    if args.browser in ["firefox", "firefox-snap"]:
+        browser_data = Path.home().joinpath(
+            "Library/Application Support/Firefox/Profiles")
+    else:
+        browser_data = Path.home().joinpath(
+            "Library/Application Support/Chrome")
 else:
-    print("Currently only Linux and macOS is supported by this script", file=sys.stderr)
+    print("Currently only Linux and macOS is supported by this script",
+          file=sys.stderr)
     sys.exit(1)
 
+cookies_path = None
+profile = args.profile
+if args.browser in ["firefox", "firefox-snap"]:
+    cookie_d_query = (
+        "SELECT value FROM moz_cookies WHERE host = '.slack.com' "
+        "AND name = 'd'"
+    )
+    cookie_ds_query = (
+        "SELECT value FROM moz_cookies WHERE host = '.slack.com' "
+        "AND name = 'd-s'"
+    )
+    if not profile:
+        profile = "*.default*"
+
+    default_profile_path = max(
+        [ x for x in browser_data.glob(profile)], key=os.path.getctime)
+    if not default_profile_path:
+        print("Couldn't find the default profile for Firefox", file=sys.stderr)
+        sys.exit(1)
+else:
+    # b'v10' is for Chromium, but not Chrome, it seems?
+    prefix =  b'v11'
```

Comment:
As you write, this is `v10` for Chromium (at least on my machine). We should check for both prefixes.

Reply:
maybe later, whenever we open another one for Chromium support